platform/windows/display_server_windows.h

```diff
@@ -468,6 +468,7 @@ class DisplayServerWindows : public DisplayServer {
 	bool use_raw_input = false;
 	bool drop_events = false;
 	bool in_dispatch_input_event = false;
+	bool last_is_double_click = false;
```

Comment:
A boolean is not enough.

Take the following sequence of mouse buttons:
1. Right MB down
2. Wait a few seconds and execute the following steps in quick succession
3. Left MB down
4. Left MB up
5. Left MB down
6. Right MB up

In step 6 the right mouse button up input event would be wrongly considered as a double click.

So this variable should rather store the mouse button index (see [`enum class MouseButton`](https://github.com/godotengine/godot/blob/7d2ca2d8ac49cde9767e00b70f9eaf1920eb266d/core/input/input_enums.h#L111))

Reply:
For reference: #89018
That PR keeps also the double-click state of different buttons separate by using `MouseButtonMask`.

Reply:
> In step 6 the right mouse button up input event would be wrongly considered as a double click.

Thanks! I have followed the instructions provided and reproduced this bug.